scripts/irods/controller.py

```diff
@@ -126,7 +126,8 @@ def start(self, write_to_stdout=False, test_mode=False):
                     cmd.append('-t')
 
                 lib.execute_command(cmd,
-                                    foreground=True,
+                                    stdout=sys.stdout,
```

Comment:
what was the foreground used for?    are we losing that functionality?

Reply:
The `foreground` argument would be passed down to `execute_command_permissive`. If set to True, it would run a loop to manually pipe the subprocess's stdout to the script's stdout.

I've removed this in favor of just passing the script's stdout directly to `subprocess.Popen`, which does the same thing without risk of blocking because of stuff like stderr filling up.

Reply:
Does the output from the Popen'd process still come out as soon as it is available? I believe the foreground option was implemented that way because the output from the irodsServer binary wasn't appearing immediately.

Reply:
I think so, but I haven't tested that specifically.
My understanding is that the `communicate` call in `communicate_and_log` blocks until the subprocess terminates, so I can force a flush after just to make sure.

Reply:
We need to verify that the output from grandpa and any forked processes is printed to the console as soon as it is available. The `--stdout` option will cause the `irodsctl` script to never return. The loop/poll/flush code allowed messages to be seen as the server was running.

Reply:
I'll do some more testing with this.

Reply:
I've done some more testing just to double check, ant it looks like the server's stdout is going directly to the console.